src/main/java/com/google/firebase/remoteconfig/Condition.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Strings;
+import com.google.firebase.internal.NonNull;
+import com.google.firebase.internal.Nullable;
+import com.google.firebase.remoteconfig.internal.TemplateResponse.ConditionResponse;
+
+import java.util.Objects;
+
+/**
+ * Represents a Remote Config condition that can be included in a {@link Template}.
+ * A condition targets a specific group of users. A list of these conditions make up
+ * part of a Remote Config template.
+ */
+public final class Condition {
+
+  private String name;
+  private String expression;
+  private TagColor tagColor;
+
+  /**
+   * Creates a new {@link Condition}.
+   *
+   * @param name A non-null, non-empty, and unique name of this condition.
+   * @param expression A non-null and non-empty expression of this condition.
+   */
+  public Condition(@NonNull String name, @NonNull String expression) {
+    this(name, expression, null);
+  }
+
+  /**
+   * Creates a new {@link Condition}.
+   *
+   * @param name A non-null, non-empty, and unique name of this condition.
+   * @param expression A non-null and non-empty expression of this condition.
+   * @param tagColor A tag color of this condition.
```

Comment:
I wonder if we might even want to clarify that this affects how the condition is displayed in the parameter list in the Firebase console.  That's true, right?  Outside the console, it is irrelevant?

Reply:
Ah kind of like you did for the setter below.  Do you agree it would be good in both places?  For creating and setting; not so much for getting I think.

   * <p>The color associated with this condition for display purposes in the Firebase Console.
   * Not specifying this value results in the console picking an arbitrary color to associate
   * with the condition.

Reply:
Good call! I updated the docs here. Thanks!